### DISCUSSION TITLE
hitting unconnected api route in Nextjs with supabase auth events (solved)

### DISCUSSION THREAD

**Discussion by csabsen** [supabase/supabase#7461] [Questions] [answered]
SOLVED: turns out messy environments do messy things and the obvious "restart everything" doesnt come to your mind because you went down the wrong rabbit hole... 
Can someone just delete this embarassment?  please 😬

Alright, I'm not turning easily to discussions but this behaviour is strange. I'm considering myself a newbie even after nearly two years of programming.

I'm implementing (since what feels like ages) the auth flow with supabase and nextjs and now I'm trying to set the supabase.auth.api.setAuthCookie  when I log in to supabase to be able to have protected routes and check for a user serverside with supabase.auth.api.getUserByCookie (and also to remove this cookie when I sign out )

When I'm logging in on the frontend this API route in nextjs (pages/api/set-auth-cookie) gets hit even though it is not connected in anyway. Is this behaviour expected?

the api route look like this (the import is just the supabase client, and the console.log to check if the route gets hit ) 
```
import { supabase } from '../../utils/supabase/supabase'

const handler = async (req, res) => {
	console.log(
		'hitting api to set auth cookie',
		new Date(Date.now()).toLocaleString(),
		req.body
	)
	supabase.auth.api.setAuthCookie(req, res)
}

export default handler
```

as a log whem logging in  I recieve this ( starred out some private stuff)
```
hitting api to set auth cookie 27/06/2022, 20:40:45 {
  event: 'SIGNED_IN',
  session: {
    access_token: '****************',
    token_type: 'bearer',
    expires_in: 3600,
    refresh_token: 'hekXs_oJFQDLPYHHtQwnrg',
    user: {
      id: 'dfddeae2-a402-471c-a3c0-5c9b50b37013',
      aud: 'authenticated',
      role: 'authenticated',
      email: '*************',
      email_confirmed_at: '2022-06-23T07:59:19.923299Z',
      phone: '',
      confirmation_sent_at: '2022-06-23T07:58:56.262578Z',
      confirmed_at: '2022-06-23T07:59:19.923299Z',
      recovery_sent_at: '2022-06-25T18:52:40.008937Z',
      last_sign_in_at: '2022-06-27T18:40:45.140428226Z',
      app_metadata: [Object],
      user_metadata: {},
      identities: [Array],
      created_at: '2022-06-23T07:58:56.257684Z',
      updated_at: '2022-06-27T18:40:45.1416Z'
    },
    expires_at: 1656358845
  }
}
```

for the user state I'm using zustand: 
```
let user = (set) => ({
	user: supabase.auth.user(),

	emailLogin: async (email, password) => {
		const { user, session, error } = await supabase.auth.signIn({
			email: email,
			password: password,
		})
		console.log('emailLogin:', { user, error, session })
		user &&
			set({
				user: user,
			})
		return { user, error }
	},

	emailSignUp: async (email, password) => {
		const { user, session, error } = await supabase.auth.signUp({
			email: email,
			password: password,
		})
		console.log('emailSignUp:', { user, error, session })
		return { user, error }
	},

	logout: async () => {
		await supabase.auth.signOut()
		set({ user: null })
	},
})

user = devtools(user, { name: 'user' })
const userStore = create(user)

export default userStore
```

Before I tried to hit this route after a SIGNED_IN event from onAuthStateChange and it would hit the API route several times (I have not counted but it was fairly unregular between 2 and 4 times), but would also explain why I would get several api hits instead of one. 

Any ideas? Grateful for any help. 

deps

```
  "dependencies": {
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "@mui/icons-material": "^5.8.0",
    "@mui/material": "^5.8.0",
    "@supabase/supabase-js": "^1.35.3",
    "axios": "^0.27.2",
    "formik": "^2.2.9",
    "formik-mui": "^4.0.0-alpha.3",
    "formik-mui-lab": "^1.0.0-alpha.3",
    "next": "12.1.6",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "yup": "^0.32.11",
    "zustand": "^4.0.0-rc.1"
  },
  "devDependencies": {
    "eslint": "8.15.0",
    "eslint-config-next": "12.1.6"
  }
}

```

**Comment by burggraf** [comment, marked as answer]
No worries.  Solving your own problems is a good thing and means you're making progress!
